nedap/formatting-stack
# Upgrade Eastwood

## Brief

Long due one!

Some of the changes in Eastwood that are most relevant to f-s:

* New reflection linter that makes our own linter redundant
* The linters that we used to disable have been thoroughly fixed and are not expected to have false positives
* Increased performance and reliability for REPL invocations
* New ways of silencing faults
  * https://github.com/jonase/eastwood#ignored-faults
  * https://github.com/jonase/eastwood#ignoring-linter-sub-kinds

## QA plan

For the Eastwood changes, I'd recommend linting large projects and verifying/fixing that the newly enabled linters aren't being problematic. Please report any false positives.

For the `runner` change, one can `(formatting-stack.core/format)` and observe that the background runner keeps doing its job.

## Author checklist



* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
    * The `runner` change is "breaking" (changed a type from Future to Thread), at the same that ns is quite clearly not an API
    * e.g. we've never received a GH issue about the runner, so presumably nobody is hacking `runner`

## Reviewer checklist

* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
